x-pack/test/api_integration/apis/ml/results/get_stopped_partitions.ts

```diff
@@ -0,0 +1,186 @@
+      }
+    });
+
+    after(async () => {
+      await ml.testResources.deleteIndexPatternByTitle('ft_module_sample_logs');
+      await ml.api.cleanMlIndices();
+    });
+
+    it('should fetch all the stopped partitions correctly', async () => {
+      const { jobId } = testSetUps[0];
+      const { body } = await supertest
+        .post(`/api/ml/results/stopped_partitions`)
+        .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+        .send({ jobIds: [jobId] })
+        .set(COMMON_REQUEST_HEADERS)
+        .expect(200);
+
+      expect(body).to.be.an('array');
+      if (body.length > 0) {
+        body.forEach((partitionValue: string) => {
+          expect(partitionValue).to.be.a('string');
+        });
+      }
+    });
+
+    it('should return empty array if stopped_on_warn is false', async () => {
+      const { jobId } = testSetUps[1];
+      const { body } = await supertest
+        .post(`/api/ml/results/stopped_partitions`)
+        .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+        .send({ jobIds: [jobId] })
+        .set(COMMON_REQUEST_HEADERS)
+        .expect(200);
+
+      expect(body).to.be.an('array');
+      expect(body).to.have.length(0);
+    });
+
+    it('should fetch stopped partitions for user with view permission', async () => {
+      const { jobId } = testSetUps[2];
+      const { body } = await supertest
+        .post(`/api/ml/results/stopped_partitions`)
+        .auth(USER.ML_VIEWER, ml.securityCommon.getPasswordForUser(USER.ML_VIEWER))
+        .send({ jobIds: [jobId] })
+        .set(COMMON_REQUEST_HEADERS)
+        .expect(200);
+
+      expect(body).to.be.an('array');
+      if (body.length > 0) {
+        body.forEach((partitionValue: string) => {
+          expect(partitionValue).to.be.a('string');
+        });
+      }
+    });
+
+    it('should not fetch stopped partitions for unauthorized user', async () => {
+      const { jobId } = testSetUps[3];
+
+      const { body } = await supertest
+        .post(`/api/ml/results/stopped_partitions`)
+        .auth(USER.ML_UNAUTHORIZED, ml.securityCommon.getPasswordForUser(USER.ML_UNAUTHORIZED))
+        .send({ jobIds: [jobId] })
+        .set(COMMON_REQUEST_HEADERS)
+        .expect(404);
+
+      expect(body.error).to.be('Not Found');
+      expect(body.message).to.be('Not Found');
+    });
+
+    it('should fetch stopped partitions for multiple job ids', async () => {
+      const { body } = await supertest
+        .post(`/api/ml/results/stopped_partitions`)
+        .auth(USER.ML_POWERUSER, ml.securityCommon.getPasswordForUser(USER.ML_POWERUSER))
+        .send({ jobIds: testJobIds })
+        .set(COMMON_REQUEST_HEADERS)
+        .expect(200);
+
+      expect(body).to.be.an('array');
+      if (body.length > 0) {
+        body.forEach((partitionValue: string) => {
+          expect(partitionValue).to.be.a('string');
```

Comment:
I'm not super happy with this validation as it would succeed also in the case that the body only contained the partition values of the first job id. So from this validation we can not tell if the multiple job ids are taken into account correctly.

Reply:
Per our discussion, I'll update the API to match the list of of unique stopped partitions to each job. So `{ job1: [], job2: [partition_1, partition_2] }` and so on for ease of use.

Reply:
Updated the API shape and tests in https://github.com/elastic/kibana/pull/74592/commits/bb698b660879faac715d5ac951a295a23795e701